### ISSUE TITLE
Add precommit hook to lint code

### ISSUE THREAD

**Issue by agustito37** [rootstrap/react-native-base#37]
Right now the lint is running in CircleCI, this isn't enough because there's a delay that diminishes the developers experience. A precommit hook of staged changes will improve this experience, we shouldn't be worried about if CircleCi is going to fail or not. 

This guy explains it pretty well: https://medium.com/@okonetchnikov/make-linting-great-again-f3890e1ad6b8

More resources:
https://github.com/react-boilerplate/react-boilerplate

**Comment by bernabe9** [comment]
CircleCI is a safe wall and it usually takes less than 15 minutes to run the linter.

I have experience using the linter with pre-commit hook and it's not safe because you can skip it using `--no-verify`.

Also, pre-commit hook could be really annoying because we create several commits in one PR and maybe we don't want to fix the lint issues in WIP commits.

I don't see any problem about using only CicleCI to check our code.

To sum up, I don't see any advantages to adding the pre-commit hook. I only see one disadvantage(mentioned before)

**Comment by agustito37** [comment]
I understand your point, and I think this ends in a preference thing, I was using it in a personal project, and really liked it. I prefer instant feedback about my code quality and don't like the idea of pushing something that has lint errors.

**Comment by bernabe9** [comment]
I'm not saying that the pre-commit hook isn't useful for some developers. As you said is a preference thing, so I think it shouldn't be in the base.

But I want to know other devs preferences

**Comment by felipegarcia92** [comment]
Can't this be a pre-push hook? Don't even know if those exist. That'd be a solution to the issue. |

And we can't control devs using `--no-verify`. It's ones' responsibility to not use it.

**Comment by agustito37** [comment]
Given that pre-commit could be a little intrusive. Using a non blocking pre-push hook could be a good idea. You push the code, even if there are errors, but you are aware of them. That's the least intrusive solution I can think of.

Something like: 
```
"hooks": {
  "pre-push": "npm run lint || exit 0"
 }
```

**Comment by felipegarcia92** [comment]
@agustito37 I think we can be strict in push. If it was per commit, then yes, I think we shouldn't block commits just because of the linter. 

But I'm pretty sure we do want clean code pushed to remote repositories.

**Comment by agustito37** [comment]
What about running the tests too?

**Comment by felipegarcia92** [comment]
@agustito37 Wouldn't add the tests. They tend to take way longer than the linter. to be run on every push.

**Comment by felipegarcia92** [comment]
Sorry, mistakenly pressed the `Close and comment` button.

**Comment by peter7z** [comment]
Closing this as it's already implemented.
